nf_core/components/install.py

```diff
@@ -50,6 +50,15 @@ def __init__(
             self.installed_by = [self.component_type]
 
     def install(self, component: str, silent: bool = False) -> bool:
+        if isinstance(component, dict):
+            if component.get("git_remote") is not None:
+                remote_url = component.get("git_remote")
```

Comment:
You've checked just above that they key exists
```suggestion
                remote_url = component["git_remote"]
```